guides/source/form_helpers.md

```diff
@@ -292,6 +292,71 @@ Some important things to notice when using `form_with` with a model object:
 * The form field names are scoped with `book[...]`. This means that `params[:book]` will be a hash containing all these field's values. You can read more about the significance of input names in chapter [Form Input Naming Conventions and Params Hash](#form-input-naming-conventions-and-params-hash) of this guide.
 * The submit button is automatically given an appropriate text value, "Create Book" in this case.
 
+You're not limited to Active Record backed models either. We can improve our
+[generic search form](#a-generic-search-form) from earlier by extracting it into
+an [Active Model](active_model_basics.html).
```

Comment:
Action View form building involves a variety of integration touch points for rendering the `<form>` element and the associated form controls.

For example, [`form_with` attempts to coerce the `:model` option](https://github.com/rails/rails/blob/8ac995453a2a2efd789810a90319e6bb6f0e2aea/actionview/lib/action_view/helpers/form_helper.rb#L769) by [invoking `#to_model` when possible](https://github.com/rails/rails/blob/13d5f87208921cc5139cd53b280c42f23c334826/actionview/lib/action_view/model_naming.rb#L6-L8). The builder uses that instance to determine a variety of attributes and options, including:

* [a default `:scope`](https://github.com/rails/rails/blob/8ac995453a2a2efd789810a90319e6bb6f0e2aea/actionview/lib/action_view/helpers/form_helper.rb#L770) based on the value returned by invoking `#param_key`
* [a default `:url` option](https://github.com/rails/rails/blob/13d5f87208921cc5139cd53b280c42f23c334826/actionview/lib/action_view/helpers/form_helper.rb#L762-L766) to be rendered to the form's `[action]` attribute (based on [polymorphic routing](https://github.com/rails/rails/blob/13d5f87208921cc5139cd53b280c42f23c334826/actionpack/lib/action_dispatch/routing/polymorphic_routes.rb#L339-L345) and any related [resolve](https://edgeapi.rubyonrails.org/classes/ActionDispatch/Routing/Mapper/CustomUrls.html#method-i-resolve) calls in the application's Routes)
* [a default `:method` option]() to be rendered to the form's `[method]` attribute (inferred from [the value of `#persisted?`](https://github.com/rails/rails/blob/13d5f87208921cc5139cd53b280c42f23c334826/actionview/lib/action_view/helpers/form_helper.rb#L1599))

Inferring default `[value]`, `[checked]`, `[selected]` attributes (or content in the case of `<textarea>` elements) for form controls involves a similar set of touch points, mainly:

* invoking methods with names that correspond to the field being rendered (for example, [determining the default `[value]` for an `<input>` element](https://github.com/rails/rails/blob/13d5f87208921cc5139cd53b280c42f23c334826/actionview/lib/action_view/helpers/tags/text_field.rb#L15) in a [type-casting compatible way](https://github.com/rails/rails/blob/main/actionview/lib/action_view/helpers/tags/base.rb#L36-L61))

While Active Model's definition of [#to_model](https://edgeapi.rubyonrails.org/classes/ActiveModel/Conversion.html), [#param_key](https://edgeapi.rubyonrails.org/classes/ActiveModel/Naming.html#method-c-param_key), [#persisted?](https://edgeapi.rubyonrails.org/classes/ActiveModel/API.html#method-i-persisted-3F), and [attribute methods exposing a variety of states](https://edgeapi.rubyonrails.org/classes/ActiveRecord/AttributeMethods/BeforeTypeCast.html#method-i-read_attribute_before_type_cast) make it a great candidate for integration, similar behavior can be achieved with plain-old Ruby objects, or objects that use underlying abstractions other than Active Model or Active Record.

Reply:
> While Active Model's definition of [#to_model](https://edgeapi.rubyonrails.org/classes/ActiveModel/Conversion.html), [#param_key](https://edgeapi.rubyonrails.org/classes/ActiveModel/Naming.html#method-c-param_key), [#persisted?](https://edgeapi.rubyonrails.org/classes/ActiveModel/API.html#method-i-persisted-3F), and [attribute methods exposing a variety of states](https://edgeapi.rubyonrails.org/classes/ActiveRecord/AttributeMethods/BeforeTypeCast.html#method-i-read_attribute_before_type_cast) make it a great candidate for integration, similar behavior can be achieved with plain-old Ruby objects, or objects that use underlying abstractions other than Active Model or Active Record.

Do you think there's a way to highlight this in the [Creating Forms with Model Objects](https://guides.rubyonrails.org/form_helpers.html#creating-forms-with-model-objects) section? The current examples are all Active Record specific, and I worry that new (and even experienced) devs won't realize they can pass a non Active Record backed object as the model.

Reply:
In order to keep this commit focused, I opened #53073